nni/compression/pytorch/speedup/compress_modules.py

```diff
@@ -3,196 +3,414 @@
 import torch
-from .infer_shape import ModuleMasks
+import torch.nn as nn
 
 _logger = logging.getLogger(__name__)
 
 replace_module = {
-    'BatchNorm2d': lambda module, mask: replace_batchnorm2d(module, mask),
-    'Conv2d': lambda module, mask: replace_conv2d(module, mask),
-    'ConvTranspose2d': lambda module, mask: replace_convtranspose2d(module, mask),
-    'MaxPool2d': lambda module, mask: no_replace(module, mask),
-    'AvgPool2d': lambda module, mask: no_replace(module, mask),
-    'AdaptiveAvgPool2d': lambda module, mask: no_replace(module, mask),
-    'ReLU': lambda module, mask: no_replace(module, mask),
-    'ReLU6': lambda module, mask: no_replace(module, mask),
-    'Sigmoid': lambda module, mask: no_replace(module, mask),
-    'Linear': lambda module, mask: replace_linear(module, mask),
-    'Dropout': lambda module, mask: no_replace(module, mask),
-    'Dropout2d': lambda module, mask: no_replace(module, mask),
-    'Dropout3d': lambda module, mask: no_replace(module, mask)
+    'BatchNorm2d': lambda module, auto_infer: replace_batchnorm2d(module, auto_infer),
+    'BatchNorm1d': lambda module, auto_infer: replace_batchnorm1d(module, auto_infer),
+    'Conv2d': lambda module, auto_infer: replace_conv2d(module, auto_infer),
+    'Linear': lambda module, auto_infer: replace_linear(module, auto_infer),
+    'MaxPool2d': lambda module, auto_infer: no_replace(module, auto_infer),
+    'AvgPool2d': lambda module, auto_infer: no_replace(module, auto_infer),
+    'AdaptiveAvgPool2d': lambda module, auto_infer: no_replace(module, auto_infer),
+    'ReLU': lambda module, auto_infer: no_replace(module, auto_infer),
+    'ReLU6': lambda module, auto_infer: no_replace(module, auto_infer),
+    'Dropout': lambda module, auto_infer: no_replace(module, auto_infer),
+    'Dropout2d': lambda module, auto_infer: no_replace(module, auto_infer),
+    'Dropout3d': lambda module, auto_infer: no_replace(module, auto_infer),
+    'LayerNorm': lambda module, auto_infer: replace_layernorm(module, auto_infer),
+    'ConvTranspose2d': lambda module, auto_infer: replace_convtranspose2d(module, auto_infer)
 }
 
+# NEED_FOLD_BIAS = True
 
-def no_replace(module, mask):
+
+def convert_to_coarse_mask(t_mask, dim):
+    """
+    Convert the mask tensor to the coarse-grained mask tensor.
+    Parameters
+    ---------
+    t_mask: torch.Tensor
+        The tensor only have 1s and 0s, 0 indicates this value is masked
+        and 1 indicates the corresponding value is not masked.
+    dim: int
+        Try to reduce the mask tensor on this dimension.
+
+    Returns
+    -------
+    indexes: torch.Tensor
+        The indexes of the sparsity that can be structurally removed.
+    remained_indexes: torch.Tensor
+        The indexes of values that need to be remained.
+    """
+    assert isinstance(t_mask, torch.Tensor)
+    shape = list(t_mask.size())
+    n_dims = len(shape)
+    dim_list = list(range(n_dims))
+    # try to reduce the mask from the 0-th dimension
```

Comment:
try to reduce the mask from the dim-th dimension

Reply:
fixed.